sway-ir/src/parser.rs

```diff
@@ -52,9 +65,24 @@ mod ir_builder {
                         ret_type,
                         locals,
                         blocks,
+                        selector
                     }
                 }
 
+            rule selector_id() -> [u8; 4]
+                = "<" _ s:$(['0'..='9' | 'a'..='f' | 'A'..='F']*<8>) _ ">" _ {
+                    let mut bytes: [u8; 4] = [0; 4];
+                    let mut cur_byte: u8 = 0;
+                    for (idx, ch) in s.chars().enumerate() {
+                        cur_byte = (cur_byte << 4) | ch.to_digit(16).unwrap() as u8;
+                        if idx % 2 == 1 {
+                            bytes[idx / 2] = cur_byte;
+                            cur_byte = 0;
+                        }
+                    }
+                    bytes
+                }
```

Comment:
I was looking at this thinking it could be done as a one liner, but the best I could come up with is:
```rust
let cnv = |c: char| c.to_digit(16).unwrap() as u8;
s.chunks(2).map(|ch| (cnv(ch[0]) << 4) | cnv(ch[1])).collect::<Vec<_>>().try_into().unwrap()
```
...which I'm not convinced is better. 🙂

Reply:
This code is mostly a duplicate of another that you wrote to parser `b256` values. I ended up merging the two in one helper function.